### ISSUE TITLE
BatchNormalization does not share parameters

### ISSUE THREAD

**Issue by Atcold** [torch/nn#282]
I need to get `nn.BatchNormalization()` to share parameters, but this is not working.

```lua
BN = nn.BatchNormalization(0)
BN2 = BN:clone('running_mean', 'running_std')
BN(torch.randn(100,5));
th> {BN}
{
  1 : 
    {
      output : DoubleTensor - size: 100x5
      gradInput : DoubleTensor - size: 100x5
      buffer : DoubleTensor - size: 100x5
      momentum : 0.1
      centered : DoubleTensor - size: 100x5
      eps : 1e-05
      train : true
      normalized : DoubleTensor - size: 100x5
      running_mean : DoubleTensor - size: 5
      buffer2 : DoubleTensor - empty
      running_std : DoubleTensor - size: 5
      std : DoubleTensor - size: 1x5
    }
}
th> {BN2}
{
  1 : 
    {
      accUpdateGradParameters : function: 0x01bb7ba8
      eps : 1e-05
      train : true
      output : DoubleTensor - empty
      gradInput : DoubleTensor - empty
      running_mean : DoubleTensor - empty
      running_std : DoubleTensor - empty
      momentum : 0.1
    }
}
```

Expected behaviour: `BN2`'s `running_mean : DoubleTensor - size: 5` and `running_std : DoubleTensor - size: 5` instead of `DoubleTensor - empty`.

**Comment by soumith** [comment]
I have a fix for this...
let me push

**Comment by Atcold** [comment]
Ehm...

**Comment by soumith** [comment]
it is clonable now, if you pull latest nn. exactly the usage that you show above in your example, will work.

**Comment by soumith** [comment]
see the docs for the updated constructor as well.

**Comment by Atcold** [comment]
OK, I didn't catch the fact there is a new constructor! Let me check!

**Comment by Atcold** [comment]
Works great, thanks!

```lua
th> BN = nn.BatchNormalization(5, nil, nil, nil)
                                                                      [0.0004s]
th> BN2 = BN:clone('running_mean', 'running_std')                                                                                                                                                                                             
                                                                      [0.0016s]
th> BN(torch.randn(100,5));
                                                                      [0.0013s]
th> {BN2}
{
  1 : 
    {
      gradBias : DoubleTensor - size: 5
      output : DoubleTensor - empty
      gradInput : DoubleTensor - empty
      accUpdateGradParameters : function: 0x08642870
      momentum : 0.1
      eps : 1e-05
      weight : DoubleTensor - size: 5
      train : true
      affine : true
      running_mean : DoubleTensor - size: 5
      bias : DoubleTensor - size: 5
      running_std : DoubleTensor - size: 5
      gradWeight : DoubleTensor - size: 5
    }
}
                                                                      [0.0005s]
th> BN2.running_mean
0.001 *
-4.2254
-9.7225
 4.1750
-4.0031
 2.9007
[torch.DoubleTensor of size 5]

                                                                      [0.0003s]
th> BN2.running_std
 1.0056
 1.0081
 0.9961
 0.9937
 0.9919
[torch.DoubleTensor of size 5]

                                                                      [0.0002s]
th> 
```
